contrib/lib/src/databases.rs

```diff
@@ -494,12 +494,12 @@ pub enum Error<T> {
     /// An error occurred while initializing an `r2d2` pool.
     Pool(r2d2::Error),
     /// An error occurred while extracting a `figment` configuration.
-    Config(figment::Error),
+    Config(Box<figment::Error>),
```

Comment:
This is unnecessary. I don't think we should box here. Sure, we'll copy a bit more, but that's fine.

Reply:
Replaced by allowing the lint with the associated comment :
```
#[allow(clippy::large_enum_variant)] // Config is only 210 bytes longer than the second longest
                                     // variant (Pool). This difference is acceptable.
```

Reply:
I don't like the idea of having these `allow(..)` spread throughout the codebase. Can we centralize them somewhere? I also don't think there's a need to justify with comments.

Reply:
Just to be clear, the goal of this PR is not to decide which `allow(...)` should be on what level. I just want Clippy's output to be clean so I can lint my project with Rocket source code in local (without telling Clippy not to lint dependencies). For the justifications, I've added them because they are very strict on this at my day job and I don't know how much comments you want. Since you're the project owner, I'll implement your preference.

To answer your question, there is a [Clippy issue](https://github.com/rust-lang/cargo/issues/5034) to add global configurations for lints. Since it is still open, the only way to centralize `alow(...)` will be to add them on top of the file. It is also possible to add the to mod.rs and lib.rs files and it will apply recursively to the submodules.

Based on your last comments, I will not justify `allow(...)` and I will add them based on this rule :
- if there is only one lint to allow in the file, I'll add it to the function;
- otherwise I'll add it at the top of the file.
Seeing how `allow(..)` are spreed will help you decide which lint should be allowed on what level. I don't think there will be that much `allow(...)` with this rule because they are mostly in the same files. I will only allow the lint [module_inception](https://rust-lang.github.io/rust-clippy/master/index.html#module_inception) in the crate's lib.rs because it is the reason why I've renamed files.

Does that way to centralize `allow(...)` suit you?